packages/back-end/src/scim/middleware/scimMiddleware.ts

```diff
@@ -0,0 +1,47 @@
+import { Response, NextFunction } from "express";
+import { getAccountPlan } from "enterprise";
+import { AuthRequest } from "../../types/AuthRequest";
+import { usingOpenId } from "../../services/auth";
+
+export default function scimMiddleware(
+  req: AuthRequest,
+  res: Response,
+  next: NextFunction
+) {
+  const acceptHeader = req.get("Accept");
+
+  // Check if the Accept header specifies SCIM JSON
+  if (acceptHeader && acceptHeader.includes("application/scim+json")) {
+    res.setHeader("Content-Type", "application/scim+json");
+  }
+
+  req.checkPermissions("manageTeam");
+
+  if (!req.organization) {
+    return res.status(400).json({
+      schemas: ["urn:ietf:params:scim:api:messages:2.0:Error"],
+      status: "400",
+      detail: "Organization is required but missing in the request.",
+    });
+  }
+
+  if (getAccountPlan(req.organization) !== "enterprise") {
+    return res.status(403).json({
+      schemas: ["urn:ietf:params:scim:api:messages:2.0:Error"],
+      status: "403",
+      detail: "SCIM requires an Enterprise GrowthBook License.",
+    });
+  }
+
+  if (!usingOpenId()) {
+    return res.status(403).json({
+      schemas: ["urn:ietf:params:scim:api:messages:2.0:Error"],
+      status: "403",
+      detail:
+        "SCIM is not available for Growthbook organizations not using SSO.",
+    });
+  }
+
+  // Continue processing the request
+  next();
```

Comment:
May want to await this and wrap in a try/catch to return a SCIM error in case one of the route handlers throws an uncaught exception.

```ts
try {
  await next();
} catch(e) {
    return res.status(400).json({
      schemas: ["urn:ietf:params:scim:api:messages:2.0:Error"],
      status: "400",
      detail: e.message,
    });
}
```

Reply:
@jdorn It doesn't look like the next() call returns a promise.

Given that, it doesn't appear the await will have an impact on the flow. Does it still make sense to have the try/catch block if we're not awaiting the next() call?

Reply:
yeah, you're right.  There are other ways to deal with async errors in Express, but not a big deal